testing/internal/e2e/tests/base/target_tcp_connect_test.go

```diff
@@ -65,6 +65,59 @@ func TestCliTcpTargetConnectTargetBasic(t *testing.T) {
 	t.Log("Successfully connected to target")
 }
 
+// TestCliTcpTargetConnectTargetBasic uses the boundary cli to create a number of
+// supporting objects to connect to a target. It then attempts to connect to
+// that target via the combination of target name and target scope name,
+// and verifies that the connection was successful.
+func TestCliTcpTargetConnectTargetViaTargetAndScopeNames(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+	c, err := loadTestConfig()
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	boundary.AuthenticateAdminCli(t, ctx)
+	newOrgId := boundary.CreateNewOrgCli(t, ctx)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary", e2e.WithArgs("scopes", "delete", "-id", newOrgId))
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	testProjectName := "E2E Project With Name"
+	testTargetName := "E2E Test Target With Name"
+	newProjectId := boundary.CreateNewProjectCli(t, ctx, newOrgId, boundary.WithName(testProjectName))
+	newHostCatalogId := boundary.CreateNewHostCatalogCli(t, ctx, newProjectId)
+	newHostSetId := boundary.CreateNewHostSetCli(t, ctx, newHostCatalogId)
+	newHostId := boundary.CreateNewHostCli(t, ctx, newHostCatalogId, c.TargetIp)
+	boundary.AddHostToHostSetCli(t, ctx, newHostSetId, newHostId)
+	newTargetId := boundary.CreateNewTargetCli(t, ctx, newProjectId, c.TargetPort, target.WithName(testTargetName))
+	boundary.AddHostSourceToTargetCli(t, ctx, newTargetId, newHostSetId)
+
+	// Connect to target via target and scope names, and print host's IP address
+	output := e2e.RunCommand(ctx, "boundary",
+		e2e.WithArgs(
+			"connect",
+			"-target-name", testTargetName,
+			"-target-scope-name", testProjectName,
```

Comment:
Connecting to target via its name and scope

Reply:
How do these options work? Do you always have to provide both options?

Reply:
Yes. The `-target-name` should be used in conjunction with either `-target-scope-name` or `-target-scope-id`

Documentation: [Selecting Targets](https://developer.hashicorp.com/boundary/docs/hcp/get-started/connect-to-target#selecting-targets)

So we may want to add a target connect attempt with `-target-name` and `-target-scope-id` as well

Reply:
Sounds good. Should we also modify the names to include some other special characters?